association/association_analysis.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import pandas as pd
+from mlxtend.frequent_patterns import apriori
+from mlxtend.frequent_patterns import association_rules
+
+
+def create_tx_map(df):
+    # (k, v): (transaction_id, list(index))
+    tx_map = {}
+    for i, cols in df.iterrows():
+        key = df['transactionId'][i]
+        tx_map.setdefault(key, [])
+        tx_map[key].append(i)
+    return tx_map
+
+
+def is_positif(x):
+    return 1 if x > 0 else 0
+
+
+def multi_dept_tx(dic, df1, c1, c2, c3, c4):
+    for k, v in dic.items():
+        if len(v) != 1:
+            for x in v:
+                c1.append(df1.iloc[x, 0])
+                c2.append(df1.iloc[x, 1])
+                c3.append(df1.iloc[x, 2])
+                c4.append(df1.iloc[x, 3])
+
+    tx_df = pd.DataFrame({'tx_id': c1, 'universe': c2, 'department': c3, 'amount': c4})
+    tx_df = tx_df[['tx_id', 'universe', 'department', 'amount']]
+    tx_df.dropna(subset=['universe'], inplace=True)
+    return tx_df
+
+
+def main():
+    ticket_df = pd.read_csv('tickets.csv', encoding='UTF-8', sep=';')
+
+    tx_id = []
+    universe = []
+    department = []
+    amount = []
+
+    tx_map = create_tx_map(ticket_df)
+
+    tx_df = multi_dept_tx(tx_map, tx_id, universe, department, amount)
+    tx_df.to_csv('multi_dept.csv', encoding='UTF-8', sep=';')
+
+    tx_df.drop(tx_df.columns[[0, 2]], axis=1, inplace=True)
+    pv_table = pd.pivot_table(tx_df, values='amount', index=['tx_id'], columns=['department'], aggfunc=np.sum)
+    pv_table.to_csv('dept_pivot_table.csv', encoding='UTF-8', sep=';', index=False)
+
+    pv_table = pv_table.applymap(is_positif)
```

Comment:
Function `is_positif` is unnecessary, you can use a lambda expression directly inline.